### DISCUSSION TITLE
Update pnp search webpart in app catalog

### DISCUSSION THREAD

**Discussion by Ainzo922** [microsoft-search/pnp-modern-search#3274] [Q&A] [answered]
Hi guys,

I have a question about updating the pnp search webpart in the app catogue in our tenant. The package allready exist but i want to update this to the latest version. What do i need to do to achieve this? Do i need to delete the existing package and update again or?

**Comment by kasperbolarsen** [comment, marked as answer]
Just overwrite the old version, a prompt will show ask if you want to keep both versions,  and you will not.
If additional permissions requests have been added to the package you will be ask to approve those.

**Reply by wobba** [reply]
...and only approve the Graph permissions you need if any. We list all the permissions the web part might need for all Graph related queries. Which depends on component usage and setup.

**Reply by Ainzo922** [reply]
Hi @kasperbolarsen and @wobba, thank you for the replies. 

I have overwritten the previous version with new package (last update) but the search webpart used in my site collections before, did not get updated to the last version. How do I push the upgrade to  all search webparts used in site collections?

**Comment by kasperbolarsen** [comment]
Hi @Ainzo922 
It depends :-)
If you have set to NOT deploy to all sites:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/6584ab7b-016d-4833-97a2-2ac69faa7904)

then you can click Gear menu->Add an App , in the Added apps section you should see PnP Modern search, Click it, and the marked button should read "Update if there is a new version available"

![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/97efe9f9-faa4-4c50-86e7-8fbd696256d6)

**Reply by Ainzo922** [reply]
Hi @kasperbolarsen, 

The webpart in my tenant is added to all sites and I have uploaded the last version but when I go to the Sharepoint site and and create page then add the pnp search webpart, I see that it’s still the old version and not the new one. So I do not know how to push the new version to the sites.

**Reply by kasperbolarsen** [reply]
It should have happened by now. Perhaps you could delete it from the App catalog and reinstall it?

**Comment by Ainzo922** [comment]
@kasperbolarsen i have some webparts active in a lot of sites so I am afraid if I delete it from the app catalog then the active ones would break? Do you have any advise?

**Comment by kasperbolarsen** [comment]
Hi @Ainzo922 

Deleting the App in the App Catalog will NOT remove the app from sites where it has already been added. So it is risk free to delete it and redeploy it.
